recipes/musl/musl_1.1.15.oe

```diff
@@ -0,0 +1,61 @@
+inherit autotools
+
+SRC_URI = "https://www.musl-libc.org/releases/musl-${PV}.tar.gz"
+
+#EXTRA_OECONF += "--enable-shared"
+RECIPE_TYPES = "machine sdk"
+
+MACHINE[unexport]       = "1"
+
+COMPATIBLE_HOST_ARCHS = ".*linux-musl.*"
+
+C_DEPENDS = "host-cross:gcc-bootstrap"
+
+DEPENDS += "linux-headers"
+DEPENDS_${PN}-dev = "linux-headers"
+RDEPENDS_${PN}-dev = "linux-headers"
+
+LIBTOOL_DEPENDS = ""
+
+inherit auto-package-libs
+AUTO_PACKAGE_LIBS = "c m rt dl pthread util crypt resolv xnet"
+
+do_install[postfuncs] += "do_install_extra_files"
+do_install_extra_files() {
+#	ln -s ../../${libdir}/libc.so ${D}${bindir}/ldd
+	for l in crypt dl m pthread resolv rt util xnet
+	do
+		ln -s libc.so ${D}${libdir}/lib$l.so
+	done
+}
+
+PACKAGES += "${PN}-ld"
+PROVIDES_${PN}-ld = "ld-so"
+PROVIDES_${PN}-ld[qa] = "is-library"
+FILES_${PN}-ld = "${base_sharedlibdir}/ld-${PV}.so ${base_sharedlibdir}/ld*.so.*"
+PACKAGES =+ "${PN}-ld-dbg"
+PROVIDES_${PN}-ld-dbg = "ld-so-dbg"
+FILES_${PN}-ld-dbg = "${base_sharedlibdir}/.debug/ld-${PV}.so"
+
+AUTO_PACKAGE_LIBS_LIBDIR = "${base_sharedlibdir}:lib:,-${PV}:.so,.so.*:"
+AUTO_PACKAGE_LIBS_LIBDIR += "${sharedlibdir}:lib:::.so"
+AUTO_PACKAGE_LIBS_LIBDIR += "${libdir}:lib:::.a,_nonshared.a,_pic.a,_pic.map"
+#AUTO_PACKAGE_LIBS:>HOST_CPU_powerpc = " nldbl"
+#AUTO_PACKAGE_LIBS_DEV_FILES .= ",_nonshared.a,_pic.a,_pic.map"
+AUTO_PACKAGE_LIBS_DEV_DEPENDS = "${PN}-dev"
+AUTO_PACKAGE_LIBS_DEV_RDEPENDS = "${PN}-dev"
+AUTO_PACKAGE_LIBS_DEPENDS = "ld-so libc"
+AUTO_PACKAGE_LIBS_RDEPENDS = "ld-so libc"
```

Comment:
As you are also auto-packaging libc, you end up with a dependency on itself for that.  I guess it has turned out to not cause problems (for now), but I don't think it is something it is worth relying on.  It might be better to add libc as explicit dependency for all those libraries needing it (even if it is all but libc itself).